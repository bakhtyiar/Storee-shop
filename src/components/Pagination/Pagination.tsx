import React from 'react';
import {useLocation, useNavigate, useParams} from 'react-router-dom';
import {Pagination as BPagination} from 'react-bootstrap';
import styled from "styled-components";

const StyledBPagination = styled(BPagination)`
	width: 100%;
	display: flex;
	justify-content: center;
	margin: 16px 0 24px 0;
`;

interface Props {
	pagesAmount: number
}

export const Pagination = ({ pagesAmount }: Props) => {
	const { page } = useParams();
	const navigate = useNavigate();
	const location = useLocation();

	const handleClick = (number: number | string) => {
		if (typeof number === "number")
			number = number.toString();
		navigate(`${location.pathname.replace(/[^/]*$/, number)}`);
	};

	return (
		<StyledBPagination data-testid={"pagination"}>

			{(Number(page) > 1) && <>

				<BPagination.First onClick={() => handleClick(1)} />

				<BPagination.Prev onClick={() => handleClick(Number(page) - 1)} />
			</>}
			{pagesAmount && [...Array(pagesAmount)]
				.map((_, i) => ++i)
				.map((number) => (
					<BPagination.Item
						key={number}
						active={number === Number(page)}
						onClick={() => handleClick(number)}
					>
						{number}
					</BPagination.Item>
				))}

			{(Number(page) !== Number(pagesAmount)) && <>

				<BPagination.Next onClick={() => handleClick(Number(page) + 1)} />

				<BPagination.Last onClick={() => handleClick(pagesAmount)} />
			</>}
		</StyledBPagination>
	);
}
