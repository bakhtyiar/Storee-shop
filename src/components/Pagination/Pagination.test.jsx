import {screen} from "@testing-library/react";
import {renderWithRouter} from "../../tests/helpers/renderWithRouter";
import {Pagination} from "./Pagination";
//todo: write interactive tests
describe("Basic tests", () => {
	test("Match snapshot", () => {
		renderWithRouter(<Pagination/>);
		expect(screen.getByTestId("pagination")).toMatchSnapshot();
	})
})