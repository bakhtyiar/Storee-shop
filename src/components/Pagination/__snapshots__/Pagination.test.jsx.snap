// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<ul
  class="sc-bcXHqe enxQNQ pagination"
  data-testid="pagination"
>
  <li
    class="page-item"
  >
    <a
      class="page-link"
      href="#"
      role="button"
      tabindex="0"
    >
      <span
        aria-hidden="true"
      >
        ›
      </span>
      <span
        class="visually-hidden"
      >
        Next
      </span>
    </a>
  </li>
  <li
    class="page-item"
  >
    <a
      class="page-link"
      href="#"
      role="button"
      tabindex="0"
    >
      <span
        aria-hidden="true"
      >
        »
      </span>
      <span
        class="visually-hidden"
      >
        Last
      </span>
    </a>
  </li>
</ul>
`;
